src/modules/launcher/Plugins/Microsoft.Plugin.Program.UnitTests/ProgramArgumentParser/ProgramArgumentParserTests.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Plugin.Program.ProgramArgumentParser;
+using NUnit.Framework;
+using Wox.Plugin;
+
+namespace Microsoft.Plugin.Program.UnitTests.ProgramArgumentParser
+{
+    [TestFixture]
+    public class ProgramArgumentParserTests
+    {
+        [TestCase("Microsoft Edge", "Microsoft Edge", null)]
+        [TestCase("Microsoft Edge ---inprivate", "Microsoft Edge ---inprivate", null)]
+        [TestCase("Microsoft Edge -- -inprivate", "Microsoft Edge", "-inprivate")]
+        [TestCase("Microsoft Edge -inprivate", "Microsoft Edge", "-inprivate")]
+        [TestCase("Microsoft Edge /inprivate", "Microsoft Edge", "/inprivate")]
+        [TestCase("edge.exe --inprivate", "edge.exe", "--inprivate")]
+        [TestCase("edge.exe -- --inprivate", "edge.exe", "--inprivate")]
+        [TestCase("edge.exe", "edge.exe", null)]
+        [TestCase("edge", "edge", null)]
+        [TestCase("cmd -c \"ping 1.1.1.1\"", "cmd", "-c \"ping 1.1.1.1\"")]
```

Comment:
i don't think this works for cmd.  It might be better to supply a real world use case `cmd /c "ping 1.1.1.1"`

Reply:
Is a small improvement yes!